otherlibs/systhreads/st_win32.h

```diff
@@ -160,53 +160,95 @@ Caml_inline void st_thread_yield(st_masterlock * m)
 
 /* Mutexes */
 
-typedef CRITICAL_SECTION * st_mutex;
+struct st_mutex_ {
+  CRITICAL_SECTION crit;
+  volatile LONG owner;    /* 0 if unlocked */
```

Comment:
```suggestion
  volatile DWORD owner;    /* 0 if unlocked */
```
(FWIW I find it more odd that a thread ID - which has type `DWORD` is stored in a "`LONG`" than that `DWORD` is passed to the Interlocked functions)
I think it would be worth an explicit comment that `owner` is **not** guarded by `crit`.

Reply:
It was `DWORD` initially, but the `Interlocked` atomic functions demand a `LONG *`, so it had to be changed to DWORD.

Agreed concerning the comment.

Reply:
Ah, fair enough! Having to cast it everywhere would be daft, I agree